.github/workflows/tests-daily-run.yml

```diff
@@ -1,7 +1,5 @@
-name: 'Daily tests run'
+name: 'Daily checks run'
```

Comment:
Don't know how much sense it makes domain-wise, but as I read the changes it looks like this workflow could trigger `.github/workflows/tests-on-demand.yml` instead.

Reply:
The plan is to remove this workflow in a followup, once I validate the nightly works as expected. I only kept it for now while transitioning to the new release workflow that will use the nightly build.

Reply:
Sounds good!